sdk/core/azure-core/inc/azure/core/http/http.hpp

```diff
@@ -454,12 +458,16 @@ namespace Azure { namespace Core { namespace Http {
     friend class Azure::Core::Test::TestHttp_getters_Test;
     friend class Azure::Core::Test::TestHttp_query_parameter_Test;
 #endif
+  public:
```

Comment:
This type is general http.  Move the typedef into Azure::Core::Http and then share it